gui/src/app/pages/HomePage/HomePage.tsx

```diff
@@ -128,79 +111,95 @@ const determineShouldBeInitiallyCollapsed = (width: number) => {
-    compiledMainJsUrl?: string
-    samplingOpts: SamplingOpts
-    setSamplingOpts: (opts: SamplingOpts) => void
+    setCompiledMainJsUrl: (url: string) => void
 }
 
-const RightView: FunctionComponent<RightViewProps> = ({ width, height, dataFileContent, saveDataFileContent, editedDataFileContent, setEditedDataFileContent, compiledMainJsUrl, samplingOpts, setSamplingOpts }) => {
+const LeftView: FunctionComponent<LeftViewProps> = ({ width, height, setCompiledMainJsUrl }) => {
+    const { data, update } = useContext(SPAnalysisContext)
     return (
         <Splitter
             direction="vertical"
             width={width}
             height={height}
-            initialPosition={height / 3}
+            initialPosition={2 * height / 3}
         >
-            <DataFileEditor
+            <StanFileEditor
                 width={0}
                 height={0}
-                fileName="data.json"
-                fileContent={dataFileContent}
-                onSaveContent={saveDataFileContent}
-                editedFileContent={editedDataFileContent}
-                setEditedFileContent={setEditedDataFileContent}
+                fileName="main.stan"
+                fileContent={data.stanFileContent}
+                // this could be made more ergonomic?
+                onSaveContent={() => update({ type: 'commitFile', filename: SPAnalysisKnownFiles.STANFILE })}
+                editedFileContent={data.ephemera.stanFileContent}
+                setEditedFileContent={(content: string) => update({ type: 'editFile', content, filename: SPAnalysisKnownFiles.STANFILE })}
                 readOnly={false}
+                setCompiledUrl={setCompiledMainJsUrl}
             />
-            <LowerRightView
+            <DataFileEditor
                 width={0}
                 height={0}
-                compiledMainJsUrl={compiledMainJsUrl}
-                dataFileContent={dataFileContent}
-                dataIsSaved={dataFileContent === editedDataFileContent}
-                samplingOpts={samplingOpts}
-                setSamplingOpts={setSamplingOpts}
+                fileName="data.json"
+                fileContent={data.dataFileContent}
+                onSaveContent={() => update({ type: 'commitFile', filename: SPAnalysisKnownFiles.DATAFILE })}
+                editedFileContent={data.ephemera.dataFileContent}
+                setEditedFileContent={(content: string) => update({ type: 'editFile', content, filename: SPAnalysisKnownFiles.DATAFILE })}
+                readOnly={false}
             />
         </Splitter>
     )
 }
 
+type RightViewProps = {
+    width: number
+    height: number
+    compiledMainJsUrl?: string
+}
+
+const RightView: FunctionComponent<RightViewProps> = ({ width, height, compiledMainJsUrl }) => {
+    return (
+        <LowerRightView
+            width={width}
+            height={height}
+            compiledMainJsUrl={compiledMainJsUrl}
+        />
+    )
+}
```

Comment:
Seems a little weird to have a `RightView` component whose sole content is the `LowerRightView` component. Are we planning to expand this in the future?

Reply:
Oops that was a remnant of the prior layout. Removed the extra component.